plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -9032,6 +9033,62 @@ public void testAutoPurgeProperty()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testExternalTablePurgeProperty() throws IOException {
+
+        TrinoFileSystem fileSystem = getTrinoFileSystem();
+        Location tempDir = Location.of("local:///temp_" + UUID.randomUUID());
+        fileSystem.createDirectory(tempDir);
+        Location dataFile = tempDir.appendPath("text.text");
+        try (OutputStream out = fileSystem.newOutputFile(dataFile).create()) {
+            out.write("helloXworld\nbyeXworld".getBytes(UTF_8));
+        }
+        // Table properties
+        String tableName = "test_external_purge_table";
+        String schema =getSession().getSchema().get();
+        StringJoiner propertiesSql = new StringJoiner(",\n   ");
+        propertiesSql.add(format("external_location = '%s'", tempDir));
+        propertiesSql.add("format = 'TEXTFILE'");
+
+        @Language("SQL") String createTableSql = format("" +
+                        "CREATE TABLE %s.%s.%s (col1 varchar,col2 varchar)\n" +
+                        "WITH (\n" +
+                        "   %s\n" +
+                        ")",
+                catalog,
+                schema,
+                tableName,
+                propertiesSql);
+        assertUpdate(createTableSql);
+
+        TableMetadata tableMetadataDefaults = getTableMetadata(catalog, schema, tableName);
+        assertThat(tableMetadataDefaults.metadata().getProperties()).doesNotContainKey(EXTERNAL_TABLE_PURGE);
+
+        assertUpdate(format("DROP TABLE %s", tableName));
+        assertThat(fileSystem.newInputFile(dataFile).exists()).isTrue(); // file should still exist after drop
+
+        propertiesSql.add("external_table_purge = true");
+        @Language("SQL") String createTableSqlWithTablePurge = format("" +
+                        "CREATE TABLE %s.%s.%s (\n" +
+                        "   col1 varchar,\n" +
+                        "   col2 varchar\n" +
+                        ")\n" +
+                        "WITH (\n" +
+                        "   %s\n" +
+                        ")",
+                catalog,
+                schema,
+                tableName,
+                propertiesSql);
+        assertUpdate(createTableSqlWithTablePurge);
+
+        TableMetadata tableMetadataWithPurge = getTableMetadata(catalog, schema, tableName);
+        assertThat(tableMetadataWithPurge.metadata().getProperties()).containsEntry(EXTERNAL_TABLE_PURGE, true);
+
+        assertUpdate(format("DROP TABLE %s", tableName));
+        fileSystem.deleteDirectory(tempDir);
```

Comment:
Please verify the empty directory to confirm `external_table_purge` effectiveness.

Reply:
This change is only about exposing a property, not made any behavioral changes related to property. Hence not verifying file deletion.

Reply:
Behavior change will be there as this property will be used by HMS, whether to delete data files or not. What i meant was, trino is not responsible for data deletion.
Trying to achieve similar behavior of [auto.purge table property](https://github.com/trinodb/trino/pull/9457)  
I have done functional testing by connecting with HMS and validated data deletion for external table.

Reply:
@electrum thoughts?

Reply:
I would recommend moving this test to `TestHive3OnDataLake`.